### ISSUE TITLE
unable to have array of objects in a parameter

### ISSUE THREAD

**Issue by AndyMoore111** [StackStorm/st2#3362]
when using the aws.autoscaling_create_launch_configuration action the BlockDeviceMappings parameter requires an array of objects (list of dictionaries) as its argument

however in the st2 web interface if you try to add a dictionary it splits on ',' and throws errors:

```Traceback (most recent call last):
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/runners/python_action_wrapper.py", line 242, in <module>
    obj.run()
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2common/runners/python_action_wrapper.py", line 154, in run
    output = action.run(**self._parameters)
  File "/opt/stackstorm/packs/aws/actions/run.py", line 20, in run
    return self.do_method(module_path, cls, action, **kwargs)
  File "/opt/stackstorm/packs/aws/actions/lib/action.py", line 140, in do_method
    resultset = getattr(obj, action)(**kwargs)
  File "/opt/stackstorm/virtualenvs/aws/lib/python2.7/site-packages/botocore/client.py", line 251, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/stackstorm/virtualenvs/aws/lib/python2.7/site-packages/botocore/client.py", line 513, in _make_api_call
    api_params, operation_model, context=request_context)
  File "/opt/stackstorm/virtualenvs/aws/lib/python2.7/site-packages/botocore/client.py", line 566, in _convert_to_request_dict
    api_params, operation_model)
  File "/opt/stackstorm/virtualenvs/aws/lib/python2.7/site-packages/botocore/validate.py", line 270, in serialize_to_request
    raise ParamValidationError(report=report.generate_report())
botocore.exceptions.ParamValidationError: Parameter validation failed:
Invalid type for parameter BlockDeviceMappings[0], value: [{"DeviceName": "/dev/sda1", type: <type 'unicode'>, valid types: <type 'dict'>
Invalid type for parameter BlockDeviceMappings[1], value: "Ebs": {"DeleteOnTermination": true, type: <type 'unicode'>, valid types: <type 'dict'>
Invalid type for parameter BlockDeviceMappings[2], value: "VolumeType": "standard", type: <type 'unicode'>, valid types: <type 'dict'>
Invalid type for parameter BlockDeviceMappings[3], value: "VolumeSize": 40}}'], type: <type 'unicode'>, valid types: <type 'dict'>
```

Entered:
```
{"DeviceName": "/dev/sda1", "Ebs": {"DeleteOnTermination": true,"VolumeType": "standard","VolumeSize": 40}}
```

If I change the type to an object:

```
[{"DeviceName": "/dev/sda1", "Ebs": {"DeleteOnTermination": true,"VolumeType": "standard","VolumeSize": 40}}]
```

the validation says 'value is not an object'

**Comment by lakshmi-kannan** [comment]
@AndyMoore111 This seems like a bug in st2web. I opened a bug there https://github.com/StackStorm/st2web/issues/352. Closing this.

**Comment by lakshmi-kannan** [comment]
Re-opening. 

@AndyMoore111 says he cannot pass array of objects even via CLI.

**Comment by bigmstone** [comment]
The CLI handles this properly. If you set type to `object` you cannot feed it an `array`. Use type `array` for that. You can set the `array` content type to `object` using something similar to:
```
parameters:
  one_happy_param:
    type: array
    items:
      type: object
    required: false
    position: 0
```

The bug; however, does exist in st2web and is fixed in st2web#360.

My assumption is you attempted to change the input to `object` and then feed it an `array` which is will get rejected when it hits st2api's validation.
